hs_file_types/models/reftimeseries.py

```diff
@@ -0,0 +1,922 @@
+                                         value=logical_file.metadata.title)
+
+    # add resource level abstract id necessary
+    if resource.metadata.description is None:
+        resource.metadata.create_element('description', abstract=logical_file.metadata.abstract)
+    # add resource level keywords
+    resource_keywords = [kw.value.lower() for kw in resource.metadata.subjects.all()]
+    for kw in logical_file.metadata.key_words:
+        if kw.lower() not in resource_keywords:
+            resource.metadata.create_element('subject', value=kw)
+
+    # add to the file level metadata
+    logical_file.metadata.keywords = logical_file.metadata.key_words
+    logical_file.metadata.save()
+
+    # add file level temporal coverage
+    start_date = min([parser.parse(series['beginDate']) for series in
+                     logical_file.metadata.serieses])
+    end_date = max([parser.parse(series['endDate']) for series in
+                    logical_file.metadata.serieses])
+    if timezone.is_aware(start_date):
+        start_date = timezone.make_naive(start_date)
+    if timezone.is_aware(end_date):
+        end_date = timezone.make_naive(end_date)
+    value_dict = {'start': start_date.isoformat(), 'end': end_date.isoformat()}
+    logical_file.metadata.create_element('coverage', type='period', value=value_dict)
+
+    # add file level spatial coverage
+    # check if we have single site or multiple sites
+    sites = set([series['site']['siteCode'] for series in logical_file.metadata.serieses])
+    if len(sites) == 1:
+        series = logical_file.metadata.serieses[0]
+        value_dict = {'east': series['site']['longitude'],
+                      'north': series['site']['latitude'],
+                      'projection': 'Unknown',
+                      'units': "Decimal degrees"}
+        logical_file.metadata.create_element('coverage', type='point', value=value_dict)
+    else:
+        bbox = {'northlimit': -90, 'southlimit': 90, 'eastlimit': -180, 'westlimit': 180,
+                'projection': 'Unknown', 'units': "Decimal degrees"}
+        for series in logical_file.metadata.serieses:
+            latitude = float(series['site']['latitude'])
+            if bbox['northlimit'] < latitude:
+                bbox['northlimit'] = latitude
+            if bbox['southlimit'] > latitude:
+                bbox['southlimit'] = latitude
+
+            longitude = float(series['site']['longitude'])
+            if bbox['eastlimit'] < longitude:
+                bbox['eastlimit'] = longitude
+
+            if bbox['westlimit'] > longitude:
+                bbox['westlimit'] = longitude
+        logical_file.metadata.create_element('coverage', type='box', value=bbox)
+
+
+def _validate_json_file(res_json_file):
+    if res_json_file.resource_file:
+        json_file_content = res_json_file.resource_file.read()
+    else:
+        json_file_content = res_json_file.fed_resource_file.read()
```

Comment:
OK. I see. Looking at ResourceFile class and I found both resource_file and fed_resource_file FileField have null and blank equal to True meaning they can both be None unless it is enforced in the code somewhere. In any case, I think it is safer to add a condition just to be sure.